### DISCUSSION TITLE
Open the chat widget from a button?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [chaskiq/chaskiq#774] [Q&A] [answered]
Would it be possible to trigger the chat widget from a button?

We have a button on our support page but don't want the float the icon on every page of the website, users can click the button to initiate this. Tawk.to has this feature for example.

**Comment by michelson** [comment, marked as answer]
yes, you can use this:
set the script embed at window.chaskiq = ...chaskiqEmbedJS

`window.chaskiq.sendCommand('toggle');`

**Reply by Jarrodsz** [reply]
Could you perhaps provide an example? 
Not totally clear how to customise the chat widget with your provided suggestion. Kind regards.

**Reply by michelson** [reply]
example: given the embed snippet
change this:

```
<script>
        (function(d,t) {
          var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
          g.src="https://app.chaskiq.io/embed.js"
          s.parentNode.insertBefore(g,s);
          g.onload=function(){
            new window.ChaskiqMessengerEncrypted({
              domain: 'https://domain',
              ws:  'wss://domain',
              app_id: "GPi9haWV1ocbr4My_Q8xPw",
              data: "YOUR_ENCRYPTED_JWE_DATA",
              lang: "USER_LANG_OR_DEFAULTS_TO_BROWSER_LANG" 
            })
          }
        })(document,"script");
      </script>
```
to this:

```
<script>
        (function(d,t) {
          var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
          g.src="https://app.chaskiq.io/embed.js"
          s.parentNode.insertBefore(g,s);
          g.onload=function(){
            window.chaskiq = new window.ChaskiqMessengerEncrypted({ // <------------- note the window.chaskiq=
              domain: 'https://domain',
              ws:  'wss://domain',
              app_id: "GPi9haWV1ocbr4My_Q8xPw",
              data: "YOUR_ENCRYPTED_JWE_DATA",
              lang: "USER_LANG_OR_DEFAULTS_TO_BROWSER_LANG" 
            })
          }
        })(document,"script");
      </script>
```

then somewhere in your code:

$(SomeElement).on('click', function(ev) {
      if (window.chaskiq) {
        ev.preventDefault();
        window.chaskiq.sendCommand('toggle'); <-- this will open the messenger if it's closed
      }
});

**Reply by Jarrodsz** [reply]
Great thank you very much. 

Only this part would require jQuery which is not recommended in a react website ( we use Gatsby )
I will try to figure out how to rewrite it to react instead.
```
$(SomeElement).on('click', function(ev) {
if (window.chaskiq) {
ev.preventDefault();
window.chaskiq.sendCommand('toggle'); <-- this will open the messenger if it's closed
}
});
```

**Reply by michelson** [reply]
the part that interests you is `window.chaskiq.sendCommand('toggle');`
